packages/widgetbook_cli/bin/command/publish.dart

```diff
@@ -0,0 +1,234 @@
+
+class PublishCommand extends WidgetbookCommand {
+  PublishCommand({
+    super.logger,
+  }) {
+    argParser
+      ..addOption(
+        'path',
+        help: 'The path to the build folder of your application.',
+        defaultsTo: './',
+      )
+      ..addOption(
+        'api-key',
+        help: 'The project specific API key for Widgetbook Cloud.',
+        mandatory: true,
+      )
+      ..addOption(
+        'branch',
+        help: 'The name of the branch for which the Widgetbook is uploaded.',
+      )
+      ..addOption(
+        'repository',
+        help:
+            'The name of the repository for which the Widgetbook is uploaded.',
+      )
+      ..addOption(
+        'commit',
+        help:
+            'The SHA hash of the commit for which the Widgetbook is uploaded.',
+      )
+      ..addOption(
+        'actor',
+        help: 'The username of the actor which triggered the build.',
+      )
+      ..addOption(
+        'git-provider',
+        help: 'The name of the Git provider.',
+        defaultsTo: 'Local',
+        allowed: [
+          'GitHub',
+          'GitLab',
+          'BitBucket',
+          'Azure',
+          'Local',
+        ],
+      )
+      ..addOption(
+        'base-branch',
+        help:
+            'The base branch of the pull-request. For example, main or master.',
+      )
+      ..addOption(
+        'base-commit',
+        help: 'The SHA hash of the commit of the base branch.',
+      )
+      ..addOption(
+        'github-token',
+        help: 'GitHub API token.',
+      )
+      ..addOption(
+        'pr',
+        help: 'The number of the PR.',
+      );
+  }
+
+  @override
+  final String description = 'Publish a new build';
+
+  @override
+  final String name = 'publish';
+
+  @override
+  Future<int> run() async {
+    final path = results['path'] as String;
+
+    if (!await GitDir.isGitDir(path)) {
+      usageException('Directory from "path" is not a Git folder');
+    }
+
+    final gitDir = await GitDir.fromExisting(
+      path,
+      allowSubdirectory: true,
+    );
+
+    final apiKey = results['api-key'] as String;
+    final currentBranch = await gitDir.currentBranch();
+    final branch = results['branch'] as String? ?? currentBranch.branchName;
+
+    final commit = results['commit'] as String? ?? currentBranch.sha;
+
+    final gitProvider = results['git-provider'] as String;
+    final gitHubToken = results['github-token'] as String?;
+    final prNumber = results['pr'] as String?;
+
+    final baseBranch = results['base-branch'] as String?;
+    late String? repository;
+    late String? actor;
+
+    if (ci.isCI && ci.Vendor.IS_GITHUB_ACTIONS) {
+      repository = Platform.environment['GITHUB_ACTION_REPOSITORY'];
+
+      actor = Platform.environment['GITHUB_ACTOR'];
+    } else {
+      repository =
+          results['repository'] as String? ?? await _getRepositoryName();
+      actor = results['actor'] as String? ?? await _getActorName();
+    }
```

Comment:
Just to understand how this works: how is it determined if a CI is e.g. Github Action?

Reply:
the packages gets info from Platform Environment  checks is Github  Actions is the current CI vendor. `Platform.environment.containsKey('GITHUB_ACTIONS');`

Reply:
It's what I've expected 👍🏻